volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -26,43 +26,113 @@
 
 
 class module(generic.GenericIntelProcess):
+    @property
+    def mod_mem_type(self):
+        """Return the mod_mem_type enum choices if available or None if not"""
+        if not self.has_member("_mod_mem_type"):
+            try:
+                vmlinux = linux.LinuxUtilities.get_module_from_volobj_type(
+                    self._context, self
+                )
+                # mod_mem_type and module_memory were added in kernel 6.4 which replaces
+                # module_layout for storing the information around core_layout etc.
+                # see commit ac3b43283923440900b4f36ca5f9f0b1ca43b70e for more information
+                self._mod_mem_type = vmlinux.get_enumeration("mod_mem_type").choices
+            except exceptions.SymbolError:
+                vollog.debug(
+                    f"Unable to find mod_mem_type enum. This is expected on kernels <6.4 but may cause issues with later kernels"
```

Comment:
This message isn't all that clear, do expect that `mod_mem_type` *doesn't* exist on kernels <6.4?  If so we should more clearly say "This error can be ignored for kernels < 6.4" or something of that nature.

Reply:
Yeah, that's the message i want to get across. On the old ones this is fine - but on something newer it probably means there is a problem. It might be better to try and read the actual kernel version and then pick a better message depending on what it is.